skimage/measure/_regionprops.py

```diff
@@ -235,7 +235,10 @@ def wrapper(obj):
         cache = obj._cache
         prop = f.__name__
 
-        if not ((prop in cache) and obj._cache_active):
+        if not obj._cache_active:
+            return f(obj)
+
+        if prop not in cache:
             cache[prop] = f(obj)
 
         return cache[prop]
```

Comment:
How about
```suggestion
        out = f(obj)
        if obj._cache_active and prop not in cache:
            cache[prop] = f(obj)
        return out
```
That logic might be even easier to follow?

Reply:
If you do that, f(obj) is computed every time, even if it's already in the cache (and cache is active).

Reply:
Oh, I entirely missed that context! Never mind then and thanks for pushing back.